pkg/daemon/ceph/osd/kms/kms.go

```diff
@@ -424,3 +464,47 @@ func SetTokenToEnvVar(ctx context.Context, clusterdContext *clusterd.Context, to
 
 	return nil
 }
+
+func put(v secrets.Secrets, secretName, secretValue string, keyContext map[string]string) error {
+	// First we must see if the key entry already exists, if it does we do nothing
+	key, err := get(v, secretName, keyContext)
+	if err != nil && err != secrets.ErrInvalidSecretId && err != secrets.ErrSecretNotFound {
+		return errors.Wrapf(err, "failed to get secret %q in kms", secretName)
+	}
+	if key != "" {
```

Comment:
Are updates supported? Or should we return an error if the value doesn't match? If the value changes it looks like the method succeeds, but ignores the new value. The caller will believe the latest value was set.

Reply:
I really don't think its supported. 
This is an existing function  used by hashcorp vault as well. The key is created for each PVC and then stored in the KMS.

Reply:
Maybe we don't need to worry about an update, since we don't expect it to change. How about logging an error if it doesn't match, but don't fail the reconcile?